iphone/Classes/TiCalendarEvent.m

```diff
@@ -418,6 +418,170 @@ -(TiCalendarRecurrenceRule*) createRecurrenceRule:(id)arg
+            {
+                [monthsOfTheYear addObject:[NSNumber numberWithInt:monthString.intValue]];
+            }
+        }
+        
+        // Weeks of the year
+        if ([component isEqualToString:@"BYWEEKNO"])
+        {
+            weeksOfTheYear = [NSMutableArray array];
+            NSArray *weekStrings = [[components objectAtIndex:++i] componentsSeparatedByString:@","];
+            for (NSString *weekString in weekStrings)
+            {
+                [weeksOfTheYear addObject:[NSNumber numberWithInt:weekString.intValue]];
+            }
+        }
+        
+        // Days of the year
+        if ([component isEqualToString:@"BYYEARDAY"])
+        {
+            daysOfTheYear = [NSMutableArray array];
+            NSArray *dayStrings = [[components objectAtIndex:++i] componentsSeparatedByString:@","];
+            for (NSString *dayString in dayStrings)
+            {
+                [daysOfTheYear addObject:[NSNumber numberWithInt:dayString.intValue]];
+            }
+        }
+        
+        // Set positions
+        if ([component isEqualToString:@"BYSETPOS"])
+        {
+            setPositions = [NSMutableArray array];
+            NSArray *positionStrings = [[components objectAtIndex:++i] componentsSeparatedByString:@","];
+            for (NSString *potitionString in positionStrings)
+            {
+                [setPositions addObject:[NSNumber numberWithInt:potitionString.intValue]];
+            }
+        }
+        
+        // RecurrenceEnd
+        if ([component isEqualToString:@"COUNT"])
+        {
+            NSUInteger occurenceCount = [[components objectAtIndex:++i] intValue];
+            recurrenceEnd = [EKRecurrenceEnd recurrenceEndWithOccurrenceCount:occurenceCount];
+            
+        } else if ([component isEqualToString:@"UNTIL"])
+        {
+            NSDate *endDate =  [dateFormatter dateFromString:[components objectAtIndex:++i]];
+            recurrenceEnd = [EKRecurrenceEnd recurrenceEndWithEndDate:endDate];
+        }
+    }
+    
+    EKRecurrenceRule* rule = [[EKRecurrenceRule alloc] initRecurrenceWithFrequency:frequency
+                                                        interval:interval
+                                                   daysOfTheWeek:daysOfTheWeek
+                                                  daysOfTheMonth:daysOfTheMonth
+                                                 monthsOfTheYear:monthsOfTheYear
+                                                  weeksOfTheYear:weeksOfTheYear
+                                                   daysOfTheYear:daysOfTheYear
+                                                    setPositions:setPositions
+                                                             end:recurrenceEnd];
+    if (rule == nil) {
+        [self throwException:@"Error while trying to create recurrance rule from RRULE string."
+            subreason: nil
+            location: CODELOCATION];
+        return nil;
+    }
+    
```

Comment:
All this creation-logic should move to the proxy, so `TiCalendarRecurrenceRule` would be renamed to `TiCalendarRecurrenceRuleProxy` and then could be accessed from outside as well. It also separates the different logical purposes better. Thx!